src/signup.js

```diff
@@ -0,0 +1,110 @@
+const emailContainer=document.querySelector('#email-input-container')
+const emailInput = emailContainer.querySelector('#email')
+const emailMsg = emailContainer.querySelector('#email-msg')
+const nameContainer=document.querySelector('#name-input-container')
+const nameInput = nameContainer.querySelector('#name')
+const nameMsg = nameContainer.querySelector('#name-msg')
+const passwordContainer=document.querySelector('#password-input-container')
+const passwordInput = passwordContainer.querySelector('#password')
+const passwordMsg = passwordContainer.querySelector('#password-msg')
+const passwordCheckContainer=document.querySelector('#password-check-input-container')
+const passwordCheckInput = passwordCheckContainer.querySelector('#password-check')
+const passwordCheckMsg = passwordCheckContainer.querySelector('#password-check-msg')
+const button = document.querySelector('#signup-button')
+
+
+emailInput.addEventListener('focusout', emailValidCheck);
+nameInput.addEventListener('focusout', nameValidCheck);
+passwordInput.addEventListener('focusout', passwordValidCheck);
+passwordCheckInput.addEventListener('focusout', passwordCheckValidCheck);
+
+emailInput.addEventListener('focusout', buttonActivate);
+nameInput.addEventListener('focusout', buttonActivate);
+passwordInput.addEventListener('focusout', buttonActivate);
+passwordCheckInput.addEventListener('focusout', buttonActivate);
+
+let emailValid=false;
+let nameValid=false;
+let passwordValid=false;
+let passwordCheckValid=false;
+
+function buttonActivate() {
+
+    if(emailValid&&nameValid&&passwordValid&&passwordCheckValid) { 
+      button.disabled = false;
+    } 
+    else { 
+      button.disabled = true; 
+    }
+}
+
+function emailValidCheck() {
+    let email=emailInput.value;
+    
+    if(email === ""){ 
+        emailValid=false;
+        emailContainer.classList.add('warning');
+        emailMsg.textContent="이메일을 입력해주세요"
+    }else if(!emailRegTest(email)) {
+        emailValid=false;
+        emailContainer.classList.add('warning');
+        emailMsg.textContent="잘못된 이메일 형식입니다"
+    }else{
+        emailValid=true;
+        emailContainer.classList.remove('warning');
+    }
+}
+
+function emailRegTest(email){     
+    //이메일 형식 검사(regular expression test)
+	regex = /^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,4}$/i;
+	if(!regex.test(email)){ 
+		return false; 
+	}else{
+		return true;
+	}
+}
```

Comment:
P1:
regex.test의 결과값이 boolean 값이므로 아래처럼 하는게 더 이해하기 쉬울 것 같습니다.

```suggestion
function emailRegTest(email){     
	const regex = /^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,4}$/i;
	return regex.test(email))
}
```